ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,277 @@
+
+## Decision
+
+### Convention over Configuration
+https://en.wikipedia.org/wiki/Convention_over_configuration
+
+Tasks should assume default locations for locating information, external systems or output storage. This aims to reduce the amount of information that needs to be configured for running a pipeline.
+
+TODO: Evaluate for the first tasks the information, which is required, and document here the defaults:
+- Resource containing the registry location (for rpm checks for instance)
+- Resource containing registry credentials
+- Resource containing the location for API scanning (for image scanning for instance)
+- Resource containing credentials for API scanning
+- Resource containing source code location (for static code analysis)
+- Etc.
+
+See: [API Contract - Create a new Build](https://docs.google.com/document/d/1IPlihVjkJ4Kb9tdhsk7iz3bn5rkT_SvJCNQhyXzK3aI/edit#heading=h.bbjyylwvwfej)
+
+### Naming convention
+It is advantageous to have naming conventions for parameters, files, and locations. Considering the multiple options: `flatcase`, `camelCase`, `PascalCase`, `dash-case`, `snake_case`, `UPPER_CASE`, `TRAIN-CASE` and the possibility to use [domain-scoped names](https://github.com/tektoncd/community/blob/main/teps/0080-support-domainscoped-parameterresult-names.md) here are the proposed naming conventions for Tasks
+
+- Parameter names: dash-case (already used in build Tasks)
+- Resource names: As per [Kubernetes recommendations](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/) resource names will follow  [RFC 1123](https://datatracker.ietf.org/doc/html/rfc1123)
+- Environment variables: Following [Kubernetes](https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/) and bash practices environment variables should be defined in UPPER_CASE
+- Labels: Labels names will follow Kubernetes practices: prefix with dot `.` as separator and dash-case for names. Prefixes and names are separated by a slash `/`. In addition to the [labels automatically created by Tekton Kubernetes standard labels or custom labels](https://tekton.dev/vault/pipelines-v0.14.3/labels/#automatically-added-labels) could be leveraged to further qualify the resources. Possible groupings (TODO: if desired, define common names):
+  - Pipeline part: Build, Test, Release, etc.
+  - Application/Product being built: AMQ Streams, EAP, RHSSO, etc.
+
+### DRY
+
+Information, which relates to an environment like (7) connection details and credentials should be configured once at the environment level and not passed as parameters for every PipelineRun. On the other hand, information, which is specific to a run like a container image digest, may be passed as a parameter.
+
+### Information injection
+
+Whenever possible, resources like ConfigMaps or Secrets will be used to inject configuration into Tasks. This is preferred to templates and patches as it fits well with Kubernetes declarative and GitOps approaches.
```

Comment:
When it comes to `ConfigMaps` how it will work with pipeline as code? I'm interested in the pre-merge case.
The problem is that in order to test changes to the ConfigMap, one should first apply them on the cluster, so we can test them in the CI.

Reply:
I don't feel we will test changes to the ConfigMap, @dirgim could you help confirm it? Thanks!

Reply:
@hongweiliu17 you're correct, we will not test changes to the `ConfigMaps` currently. 
We're currently only using `Secrets` to inject configuration that allows tasks to access the testing environments and are not using `ConfigMaps` at this point, but have left them as an option for future enhancements.

Reply:
Thanks Kruno.